### PR TITLE
Remove dependency to R.Components from Languages.Editor

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

**Comment by MikhailArkhipov** [comment]
Please don't merge until 1.0 is forked

**Comment by AlexanderSher** [comment]
Sure, that is for CR only.

File: src/Common/Core/Impl/UI/Commands/CommandStatus.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 using System;
 
-namespace Microsoft.R.Components.Controller {
+namespace Microsoft.Common.Core.UI.Commands {
```
Comment (MikhailArkhipov) [inline src/Common/Core/Impl/UI/Commands/CommandStatus.cs]: Probably just 'Commands' since controllers are not necessarily UI related.

Comment (AlexanderSher) [inline src/Common/Core/Impl/UI/Commands/CommandStatus.cs]: `IAsyncCommand` has `Status` property of `CommandStatus` enum, which has `Invisible` and `Latched` values that are UI-related.

File: src/Languages/Editor/Impl/Services/ServiceManager.cs [hunk 2]
```diff
@@ -48,11 +62,13 @@ public sealed class ServiceManager : ServiceManagerBase {
             sm.ServiceRemoved += onServiceRemoved;
         }
 
-        private ServiceManager(IPropertyOwner propertyOwner, ICoreShell coreShell) : base(propertyOwner) {
+        private ServiceManager(IPropertyOwner propertyOwner, ICoreShell coreShell) {
```
Comment (MikhailArkhipov) [inline src/Languages/Editor/Impl/Services/ServiceManager.cs]: While we at this - I think IExportProvider would be sufficient

File: src/Package/Impl/Commands/AsyncCommandController.cs [hunk 3]
```diff
@@ -4,8 +4,11 @@
...
 using System.Collections.Generic;
 using Microsoft.Common.Core;
+using Microsoft.Common.Core.UI.Commands;
+using Microsoft.R.Components.Controller;
+using static System.FormattableString;
 
-namespace Microsoft.R.Components.Controller {
+namespace Microsoft.VisualStudio.R.Package.Commands {
     public class AsyncCommandController : ICommandTarget {
         private Dictionary<Key, IAsyncCommand> CommandMap { get; } = new Dictionary<Key, IAsyncCommand>();
```
Comment (MikhailArkhipov) [inline src/Package/Impl/Commands/AsyncCommandController.cs]: probably doesnt need to be property

File: src/R/Components/Impl/ConnectionManager/Commands/ReconnectCommand.cs [hunk 4]
```diff
@@ -21,12 +21,12 @@ public class ReconnectCommand : IAsyncCommand {
             ? CommandStatus.Supported
             : CommandStatus.SupportedAndEnabled;
 
-        public Task<CommandResult> InvokeAsync() {
```
Comment (MikhailArkhipov) [inline src/R/Components/Impl/ConnectionManager/Commands/ReconnectCommand.cs]: You don't want result? It is generally permitted just to call Invoke without actually checking status first.

File: src/R/Components/Impl/Help/Commands/HelpHomeCommand.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+using Microsoft.Common.Core.Shell;
+using Microsoft.Common.Core.UI.Commands;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.R.Host.Client;
+
+namespace Microsoft.R.Components.Help.Commands {
+    public sealed class HelpHomeCommand : IAsyncCommand {
+        private readonly ICoreShell _shell;
+
+        public HelpHomeCommand(ICoreShell shell) {
```
Comment (MikhailArkhipov) [inline src/R/Components/Impl/Help/Commands/HelpHomeCommand.cs]: May just be IExportProvider I think

File: src/R/Components/Impl/Microsoft.R.Components.csproj [hunk 6]
```diff
@@ -390,7 +383,10 @@
   <ItemGroup>
     <Resource Include="Resources\packageicon.png" />
   </ItemGroup>
-  <ItemGroup />
+  <ItemGroup>
```
Comment (MikhailArkhipov) [inline src/R/Components/Impl/Microsoft.R.Components.csproj]: Empty folders?

File: src/Package/Impl/Commands/AsyncCommandController.cs (hunk 3 above)
Comment (AlexanderSher) [inline src/Package/Impl/Commands/AsyncCommandController.cs]: Yep

File: src/Languages/Editor/Impl/Services/ServiceManager.cs (hunk 2 above)
Comment (AlexanderSher) [inline src/Languages/Editor/Impl/Services/ServiceManager.cs]: `ICoreShell` is required for `IdleTimeAction.Create`

File: src/R/Components/Impl/Help/Commands/HelpHomeCommand.cs (hunk 5 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/Help/Commands/HelpHomeCommand.cs]: Yep

File: src/R/Components/Impl/Microsoft.R.Components.csproj (hunk 6 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/Microsoft.R.Components.csproj]: Yeah, should be deleted.

File: src/R/Components/Impl/ConnectionManager/Commands/ReconnectCommand.cs (hunk 4 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/ConnectionManager/Commands/ReconnectCommand.cs]: I mean, if we want to have real async command execution, we could add `InvokeAsync` and `GetStatusAsync` or make  new `ICommandTargetAsync` and do `.Wait` only when sending down to OLE or base editor. However, OLE shim and base calls would have to transition to UI thread.  So up to you.

**Comment by MikhailArkhipov** [comment]
👍

**Pull request closed without merging**